### PR TITLE
feat: add open flag to dfx deploy command

### PR DESCRIPTION
**Pull request by filippofinke**
# Description

This pull request adds a new feature to the dfx deploy command where users can specify the --open flag to automatically open the URL of the deployed canisters in the default browser of the machine. This enhancement aims to improve user experience by providing a convenient way to access the deployed canisters immediately after deployment.

# How Has This Been Tested?

To test this feature:

- Ensure the dfx command-line tool is installed and configured properly.
- Run the dfx deploy --open command with a project that has deployable canisters.
- Verify that after deployment, the default browser opens with the URL of the deployed canisters.

# Checklist:

- [x] The title of this PR complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I have edited the CHANGELOG accordingly.
- [x] I have made corresponding changes to the documentation.

### CODE REVIEW

**Comment by sa-github-api** [comment]
Dear @filippofinke,

In order to potentially merge your code in this open-source repository and therefore proceed with your contribution, we need to have your approval on DFINITY's [CLA](https://github.com/dfinity/cla/blob/main/CLA.md).

If you decide to agree with it, please visit this [issue](https://github.com/dfinity/cla/issues/451) and read the instructions there. Once you have signed it, re-trigger the workflow on this PR to see if your code can be merged.

— The DFINITY Foundation

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,27 +2,32 @@
...
+### feat: add --open flag to dfx deploy
+
+The `--open` flag will open the browser to the canister's URL after deployment.
+
 ### feat: unify CLI options to specify arguments
 
 There are a few subcommands that take `--argument`/`--argument-file` options to set canister call/init arguments.
 
 We unify the related logic to provide consistent user experience.
-
```
Comment (ericswanson-dfinity) [inline CHANGELOG.md:10]: Please revert all of the changes to this file that are not in the "UNRELEASED" section.

Comment (filippofinke) [reply CHANGELOG.md]: Done!

File: src/dfx/src/commands/deploy.rs [hunk 2]
```diff
@@ -248,13 +252,25 @@ fn display_urls(env: &dyn Environment) -> DfxResult {
         if !frontend_urls.is_empty() {
             info!(log, "  Frontend canister via browser");
             for (name, url) in frontend_urls {
-                info!(log, "    {}: {}", name, green.apply_to(url));
+                info!(log, "    {}: {}", name, green.apply_to(&url));
+
+                if open_in_browser {
+                    if let Err(e) = webbrowser::open(url.as_str()) {
+                        info!(log, "Failed to open browser: {}", e);
```
Comment (ericswanson-dfinity) [inline src/dfx/src/commands/deploy.rs:259]:
```suggestion
                        error!(log, "Failed to open browser: {}", e);
```

File: src/dfx/src/commands/deploy.rs [hunk 3]
```diff
@@ -248,13 +252,25 @@ fn display_urls(env: &dyn Environment) -> DfxResult {
...
         }
         if !candid_urls.is_empty() {
             info!(log, "  Backend canister via Candid interface:");
             for (name, url) in candid_urls {
-                info!(log, "    {}: {}", name, green.apply_to(url));
+                info!(log, "    {}: {}", name, green.apply_to(&url));
+
+                if open_in_browser {
+                    if let Err(e) = webbrowser::open(url.as_str()) {
+                        info!(log, "Failed to open browser: {}", e);
```
Comment (ericswanson-dfinity) [inline src/dfx/src/commands/deploy.rs:271]:
```suggestion
                        error!(log, "Failed to open browser: {}", e);
```

**Pull request closed without merging**